REPL.cpp

```diff
@@ -31,18 +30,44 @@ swift::FuncDecl *ConfigureFunctionLinkage(swift::SourceFile &src_file, std::uniq
 
             sil_module->lookUpFunction(sil_decl)->setLinkage(swift::SILLinkage::Public);
             Log(std::string("Set function ") + name_original + " (" + name_mangled + ") to public");
-
-            if(name_original == res_fn_name)
-                res_fn = fn;
         }
     }
     sil_module->lookUpFunction("main")->setLinkage(swift::SILLinkage::Private);
-    return res_fn;
 }
 
-llvm::Expected<std::unique_ptr<REPL>> REPL::Create()
+void REPL::RemoveRedeclarationsFromJIT(std::unique_ptr<llvm::Module> &llvm_module)
 {
-    std::unique_ptr<REPL> result(new REPL);
+    SetCurrentLoggingArea(LoggingArea::SIL);
+
+    for(llvm::Function &fn : llvm_module->functions())
+    {
+        if(fn.isDeclaration() || !fn.hasDefaultVisibility())
+            continue;
+        std::string name = fn.getName().str();
+        llvm::consumeError(m_jit->RemoveSymbol(name));
```

Comment:
No error checking? llvm::Success would be fine to continue on, but what if something else goes wrong? :P

Reply:
Well the way I see it, it returns Success if it successfully removed it, and an actual error if the symbol isn't found. Either way, that symbol doesn't exist, in the JIT, so we can continue.

Reply:
What if the JIT runs into an unrelated issue?

Reply:
Well the function I'm calling (`JITDylib::remove`), which can only return two errors:
```
If any symbols are not defined in this JITDylib this method will return a SymbolsNotFound error covering the missing symbols.

If all symbols are found but some symbols are in the process of being materialized this method will return a SymbolsCouldNotBeRemoved error.
```
Both can be safely ignored (I'll add a comment though)

Reply:
Now that I think about it, the second one probably should never happen, so i'll handle it.